outwatch/src/main/scala/outwatch/Handler.scala

```diff
@@ -9,27 +10,37 @@ import monix.reactive.{Observable, Observer}
 import scala.concurrent.Future
 
 object Handler {
-  def empty[T]:IO[Handler[T]] = create[T]
+  def empty[F[_]: Sync, T]: F[Handler[T]] = create[F, T]
 
-  def create[T]:IO[Handler[T]] = IO(unsafe[T])
-  def create[T](seed:T):IO[Handler[T]] = IO(unsafe[T](seed))
+  def create[F[_]] = new CreatePartiallyApplied[F]
+  def create[F[_], T](implicit F: Sync[F]): F[Handler[T]] = F.delay(unsafe[T])
+  def create[F[_], T](seed: T)(implicit F: Sync[F]): F[Handler[T]] = F.delay(unsafe[T](seed))
```

Comment:
I am wondering, whether we should additionally give users a default effect to use with `Handler.create`. So they do not need to think which effect they should use in each case. The current generic ones are nice, when you want to mix them with other effects in a for-comprehension. But if you are just doing `Handler.create[Int].map(handler => div(handler, ...))`, you do not really need to worry which effect type is used -- defaulting to SyncIO here would be a good choice performance-wise.

Should we maybe have another set of methods where you do not need to provide the effect type, for example:

```scala
def apply[T] = create[SyncIO, T]
def apply[T](seed: T) = create[SyncIO, T](seed)
```

Not sure about the naming, whether it should be `apply` or maybe `create` and `createF`. What do you think?

Reply:
Any opinions? :)